Lombiq.HelpfulExtensions/Assets/Scripts/target-blank.js

```diff
@@ -5,16 +5,11 @@
 
         for (let i = 0; i < links.length; i++) {
             if (!links[i].href.match(/^mailto:/) &&
-                (links[i].hostname !== currentHostname &&
-                (!links[i].href.match(/^javascript:/i)))) {
+                links[i].hostname !== currentHostname &&
+                !links[i].href.match(/^javascript:/i)) {
```

Comment:
These checks can be ordered and simplified, discarding the regex usage and simplifying the whole expression as follows:

```c#
            if (links[i].hostname !== currentHostname &&
                !links[i].href.startsWith('javascript:') &&
                !links[i].href.startsWith('mailto:')) {
```

Reply:
Analyzer error: Script URL is a form of eval. At line 8: !links[i].href.startsWith('javascript:') &&

Reply:
Please link the error in the log. Didn't you get that error locally?

Reply:
https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/4232004950

![image](https://user-images.githubusercontent.com/101474974/220343233-d8b90fb2-8e7e-4bf3-b736-e8c0a4df7688.png)

Locally there are no problems/errors, only GHA throws this.

@0liver

Reply:
This error did come up earlier, when working on OSOE-476, thats why i changed the script to its current form, meaning this:
links[i].hostname !== currentHostname &&
                !links[i].href.match(/^javascript:/i)) {

Reply:
This would be a working solution, with the requested reordering:
![image](https://user-images.githubusercontent.com/101474974/220344749-d57055fe-64ac-412c-b046-42d78ed5c879.png)

If you accept it.

Reply:
FYI, you can link to individual lines in the log like this:

![image](https://user-images.githubusercontent.com/164360/220344674-50cc1b58-7244-424c-bea9-8f87a7509840.png)

1. I'm surprised you don't see any errors locally because you should.
2. Tricking the analyzer is definitely **_not_** the desired way to resolve the warning.

Instead, disable the warning for this concrete use case **because it refers to code that is not our own**. The [no-script-url](https://eslint.org/docs/latest/rules/no-script-url) rule triggers to prevent the usage of `javascript:` URLs. In the current script, we are not using them - the script is meant to run on 3rd-party pages that we may not control. This is a case where suppressing the warning and adding an explanatory comment to the suppression is the correct approach.

You'll find documentation on how exactly to suppress linter warnings in the NE docs.